brewtils/__init__.py

```diff
@@ -66,46 +119,64 @@ def get_connection_info(cli_args=None, **kwargs):
 
-    Configuation will be loaded from these sources, with earlier sources having higher priority:
+    Configuration will be loaded from these sources, with earlier sources having
+    higher priority:
 
         1. ``**kwargs`` passed to this method
         2. ``cli_args`` passed to this method
         3. Environment variables using the ``BG_`` prefix
         4. Default values in the brewtils specification
 
     Args:
-        cli_args (list, optional): List of command line arguments for configuration loading
+        cli_args (list, optional): List of command line arguments for
+            configuration loading
+        argument_parser (ArgumentParser, optional): Argument parser to use when
+            parsing cli_args. Supplying this allows adding additional arguments
+            prior to loading the configuration. This can be useful if your
+            startup script takes additional arguments.
+        merge_specification (dict, optional): Specification that will be merged
+            with the brewtils specification before loading the configuration
         **kwargs: Additional configuration overrides
 
     Returns:
         :obj:`box.Box`: The resolved configuration object
     """
+    spec_source = merge_specification or {}
+    spec_source.update(SPECIFICATION)
+    spec = YapconfSpec(spec_source, env_prefix='BG_')
+
     sources = []
 
     if kwargs:
         # Do a little kwarg massaging for backwards compatibility
         if 'bg_host' not in kwargs and 'host' in kwargs:
-            warnings.warn("brewtils.load_config called with 'host' keyword argument. This name "
-                          "will be removed in version 3.0, please use 'bg_host' instead.",
+            warnings.warn("brewtils.load_config called with 'host' keyword "
+                          "argument. This name will be removed in version 3.0, "
+                          "please use 'bg_host' instead.",
                           DeprecationWarning, stacklevel=2)
             kwargs['bg_host'] = kwargs.pop('host')
         if 'bg_port' not in kwargs and 'port' in kwargs:
-            warnings.warn("brewtils.load_config called with 'port' keyword argument. This name "
-                          "will be removed in version 3.0, please use 'bg_port' instead.",
+            warnings.warn("brewtils.load_config called with 'port' keyword "
+                          "argument. This name will be removed in version 3.0, "
+                          "please use 'bg_port' instead.",
                           DeprecationWarning, stacklevel=2)
             kwargs['bg_port'] = kwargs.pop('port')
 
         sources.append(('kwargs', kwargs))
 
     if cli_args:
-        from argparse import ArgumentParser
+        argument_parser = argument_parser or ArgumentParser()
 
-        parser = ArgumentParser()
-        spec.add_arguments(parser)
-        parsed_args = parser.parse_args(cli_args)
+        spec.add_arguments(argument_parser)
```

Comment:
This will break if you're actually using the argument parser returned by our `get_argument_parser` method. You'll get something like:

```python
parser = get_argument_parser()
parser.add_argument('positional')
load_config(parser=parser)
argparse.ArgumentError: argument --bg-host: conflicting option string: --bg-host
```

Reply:
Fixed